apis/v1alpha1/image_policy.go

```diff
@@ -227,6 +228,9 @@ var filterValidationSchema = v1beta1.JSONSchemaProps{
 		{
 			Required: []string{"github"},
 		},
+		{
+			Required: []string{"pinned"},
```

Comment:
So with the `OneOf`, we can make this more interesting. We can now also add other JSONSchemaProps, so we can add a `Properties` value here, which then again has a `Required` field with `version` required in it.

This way we make sure that if someone chooses `pinned`, they also set a `version`.